hazelcast/src/main/java/com/hazelcast/client/impl/statistics/ClientStatisticsService.java

```diff
@@ -128,28 +134,30 @@ private TcpClientConnection getConnection() {
      * @param periodSeconds the interval at which the statistics collection and send is being run
      */
     private void schedulePeriodicStatisticsSendTask(long periodSeconds) {
-        ClientMetricCollector clientMetricCollector = new ClientMetricCollector();
-        CompositeMetricsCollector compositeMetricsCollector = new CompositeMetricsCollector(clientMetricCollector,
+        clientMetricCollector = new ClientMetricCollector();
+        compositeMetricsCollector = new CompositeMetricsCollector(clientMetricCollector,
```

Comment:
I don't think these classes are thread-safe. We are now using the same instances possibly across different threads (imagine you have the regular scheduled statistics collection and some new collection due to cluster change running at the same time). I believe we should revert making them instance fields